twine/__main__.py

```diff
@@ -25,7 +27,23 @@ def main() -> Any:
     try:
         return cli.dispatch(sys.argv[1:])
     except (exceptions.TwineException, requests.HTTPError) as exc:
-        return "{}: {}".format(exc.__class__.__name__, exc.args[0])
+        return _format_error(f"{exc.__class__.__name__}: {exc.args[0]}")
+
+
+def _format_error(err: str) -> str:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--no-color", default=False, required=False, action="store_true",
+    )
+
+    args, _ = parser.parse_known_args(sys.argv[1:])
```

Comment:
See ongoing conversation in https://github.com/pypa/twine/pull/649#discussion_r436357825.

Regardless of the outcome of that, I don't think the `parser` logic belongs here.